runtime/near-vm-runner/src/cache.rs

```diff
@@ -85,6 +87,110 @@ impl fmt::Debug for MockCompiledContractCache {
     }
 }
 
+struct FilesystemCompiledContractCacheState {
+    dir: rustix::fd::OwnedFd,
+    test_temp_dir: Option<tempfile::TempDir>,
+}
+
+#[derive(Clone)]
+pub struct FilesystemCompiledContractCache {
+    state: Arc<FilesystemCompiledContractCacheState>,
+}
+
+impl FilesystemCompiledContractCache {
+    pub fn new<HD: AsRef<std::path::Path> + ?Sized, SP: AsRef<std::path::Path> + ?Sized>(
+        home_dir: &HD,
+        store_path: Option<&SP>,
+    ) -> std::io::Result<Self> {
+        let path = std::env::var_os("NEAR_COMPILED_CACHE");
+        let path = path.map(PathBuf::from).unwrap_or_else(|| {
+            let store_path = store_path.map(AsRef::as_ref).unwrap_or_else(|| "data".as_ref());
+            home_dir.as_ref().join(store_path).join("contracts")
+        });
```

Comment:
I don't see why do we need both env and config. Can you pick one (config :)) and just use this one? 
If there is a good reason for having both can you add a small comment?